src/org/zaproxy/zap/extension/pscanrulesAlpha/XPoweredByHeaderInfoLeakScanner.java

```diff
@@ -58,30 +59,51 @@ public void scanHttpRequestSend(HttpMessage msg, int id) {
 	@Override
 	public void scanHttpResponseReceive(HttpMessage msg, int id, Source source) {
 		long start = System.currentTimeMillis();
-	
 		Vector<String> xpbOptions = msg.getResponseHeader().getHeaders(HEADER_NAME);
-		if (xpbOptions != null) { //Header Found
-			for (String xpbDirective : xpbOptions) {
-				Alert alert = new Alert(getPluginId(), Alert.RISK_LOW, Alert.CONFIDENCE_MEDIUM, //PluginID, Risk, Reliability
-					getName()); 
-		    		alert.setDetail(
-		    				getDescription(), //Description
-		    				msg.getRequestHeader().getURI().toString(), //URI
-		    				"",	// Param
-		    				"", // Attack
-		    				"", // Other info
-		    				getSolution(), //Solution
-		    				getReference(), //References
-		    				HEADER_NAME, // Evidence - Return the header name
-		    				200, // CWE Id
-		    				13,	// WASC Id
-		    				msg); //HttpMessage
-		    	parent.raiseAlert(id, alert);
-				}
-			}
-		    if (logger.isDebugEnabled()) {
+        if (xpbOptions != null) {
+            raiseAlert(msg, id, xpbOptions);
+        }
+		if (logger.isDebugEnabled()) {
 		    	logger.debug("\tScan of record " + id + " took " + (System.currentTimeMillis() - start) + " ms");
-	    }
+		}
+	}
+
+    /**
+     * Raises an alert with the "Evidence" or "Other" field filled-in depending on the header repetition.
+     * @param msg
+     * @param id
+     */
+	private void raiseAlert(HttpMessage msg, int id, List<String> optionValues) {
+        String keyValueOptionForEvidenceField = "";
+        String keyValueOptionForOtherField = "";
+        if (optionValues.size() == 1) {
+            keyValueOptionForEvidenceField = HEADER_NAME + ": " + optionValues.get(0);

```

Comment:
Yes, that's about the matching/highlighting not working correctly when the header name has a different case.

It does not require changes in core, it can be obtained using a regex of the header name and the corresponding value. [1]
Of course, it should be possible to obtain the whole header (instead of just the value) using the `HttpHeader` class.

[1] https://github.com/zaproxy/zap-extensions/blob/7658674921d42850ff0bbe81a5acd7a0e1847f66/src/org/zaproxy/zap/extension/pscanrules/SetCookieUtils.java#L127-L145